pyppl/proc.py

```diff
@@ -1,1053 +1,363 @@
-		# The jobs
-		self.props.jobs = []
-		# The locker for the process
-		self.props.lock = None
-		# non-cached job ids
-		self.props.ncjobids = []
-		# The original name of the process if it's copied
-		self.props.origin = defaultconfig['id']
-		# The computed output
-		self.props.output = OrderedDiot()
-		# data for proc.xxx in template
-		self.props.procvars = {}
-		# Valid return code
-		self.props.rc = [0]
-		# get the runner from the profile
-		self.props.runner = 'local'
-		# The computed script. Template object
-		self.props.script = None
-		# The unique identify of the process
-		# cache the suffix
-		self.props._suffix = ''
-		# The template class
-		self.props.template = None
-		# timer for running time
-		self.props.timer = None
-		# The computed workdir
-		self.props.workdir = ''
-		# Remember the attr being set, they have the highest priority
-		self.props.sets = set()
-
-		# convert alias to its original name
-		for aliaskey, aliasval in Proc.ALIAS.items():
-			if aliaskey in kwargs:
-				kwargs[aliasval] = kwargs.pop(aliaskey)
-
-		for key in kwargs:
-			if key not in defaultconfig:
-				raise ProcAttributeError(key)
-
-		# update the conf with kwargs
-		defaultconfig.update(kwargs)
-		# collapse the loading trace, we don't need it anymore.
-		self.config._load({'default': defaultconfig})
-		for key, val in kwargs.items():
-			if key[0] != '_':
-				setattr(self, key, val)
-
-	def __getattr__(self, name):
-		"""
-		Get the value of a property in `self.props`
-		It recognizes alias as well.
-		@params:
-			`name`: The name of the property
-		@returns:
-			The value of the property
-		"""
-		if name not in self.props \
-			and name not in self.config \
-			and name not in Proc.ALIAS \
-			and not name.endswith ('Runner'):
-			raise ProcAttributeError(name)
-
-		if name in self.props or name in self.config:
-			pass
-		elif name in Proc.ALIAS:
-			name = Proc.ALIAS[name]
-
-		ret = pluginmgr.hook.procGetAttr(proc = self, name = name) # pylint: disable=no-member
-		if ret is None:
-			return self.props.get(name, self.config.get(name))
-
-		return ret
-
-	def __setattr__(self, name, value): # pylint: disable=too-many-branches,too-many-statements
-		"""
-		Set the value of a property in `self.config`
-		@params:
-			`name` : The name of the property.
-			`value`: The new value of the property.
-		"""
-		if name not in self.config and name not in Proc.ALIAS \
-			and not name.endswith ('Runner'):
-			raise ProcAttributeError(name, 'Cannot set attribute for process')
-
-		# profile will be deprecated, use runner instead
-		# currently nothing deprecated
-		if name in Proc.DEPRECATED: # pragma: no cover
-			logger.warning('Attribute "%s" is deprecated%s.', name,
-				', use "{}" instead'.format(Proc.DEPRECATED[name]) \
-					if name in Proc.DEPRECATED and Proc.DEPRECATED[name] else '',
+			runtime_config (simpleconf.Config): The runtime configuration
+		"""
+		self.runtime_config = runtime_config
+		self.input
+		self.output
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Statement seems to have no effect](https://app.codacy.com/app/pwwang/PyPPL/pullRequest?prid=4727008)